### ISSUE TITLE
ExtensionLoader exception after upgrade

### ISSUE THREAD

**Issue by cpinfold** [eventum/eventum#522]
**Version(s) affected**: 3.6.5

**Description**  
After upgrade to 3.6.5 from 3.5.6, ExtensionLoader is throwing an exception due to L145 returning null which doesn't match declared return type of string for getClassName.

Uncaught Exception TypeError: "Return
value of Eventum\Extension\ExtensionLoader::getClassName() must be of
the type string, null returned" at
/var/www/support/src/Extension/ExtensionLoader.php line 145

**How to reproduce**  

**Possible Solution**  

**Additional context**

**Comment by glensc** [comment]
I'm sure this is fixed already in next versions!

**Comment by cpinfold** [comment]
Thanks.  I only checked master before reporting.

https://github.com/eventum/eventum/blob/40a6de2fc8c97ae58040434fa90237d5974b92d4/src/Extension/ExtensionLoader.php#L145

Should be 
`return '';`

**Comment by glensc** [comment]
ok, based on your linked code, seems it's broken in master as well. please provide scenario how to reproduce this.

**Comment by cpinfold** [comment]
I had just upgraded to 3.6.5 (from 3.5.6)
I was trying to run 'php bin/upgrade.php'
Note: I do have a workflow class which was copied into the upgraded config.
````
[2019-04-15 00:29:32] app.ERROR: Uncaught Exception TypeError: "Return
value of Eventum\Extension\ExtensionLoader::getClassName() must be of
the type string, null returned" at
/var/www/PATH/src/Extension/ExtensionLoader.php line 145
{"exception":"[object] (TypeError(code: 0): Return value of
Eventum\\Extension\\ExtensionLoader::getClassName() must be of the
type string, null returned at
/var/www/PATH/src/Extension/ExtensionLoader.php:145)
[stacktrace]
#0 /var/www/PATH/src/Extension/ExtensionLoader.php(90):
Eventum\\Extension\\ExtensionLoader->getClassName('.class.WORKFLOW....')
#1
/var/www/PATH/src/Extension/BuiltinLegacyLoaderExtension.php(88):
Eventum\\Extension\\ExtensionLoader->getClassList()
#2
/var/www/PATH/src/Extension/BuiltinLegacyLoaderExtension.php(47):
Eventum\\Extension\\BuiltinLegacyLoaderExtension->createAutoloadMap(Object(Eventum\\Extension\\ExtensionLoader),
Array)
#3 /var/www/PATH/src/Extension/ExtensionManager.php(166):
Eventum\\Extension\\BuiltinLegacyLoaderExtension->registerAutoloader(Object(Composer\\Autoload\\ClassLoader))
#4 /var/www/PATH/src/Extension/ExtensionManager.php(47):
Eventum\\Extension\\ExtensionManager->loadExtensions()
#5 /var/www/PATH/src/Extension/ExtensionManager.php(39):
Eventum\\Extension\\ExtensionManager->__construct()
#6 /var/www/PATH/init.php(116):
Eventum\\Extension\\ExtensionManager::getManager()
#7 /var/www/PATH/bin/upgrade.php(28):
require_once('/var/www/PATH...')
#8 {main}
"}
{"file":null,"line":null,"class":null,"function":null,"memory_usage":"4
MB","memory_peak_usage":"4 MB","version":"3.6.5","usr_id":""}
````

**Comment by glensc** [comment]
@cpinfold please put yourself to my perspective. how do I reproduce this? your bugreport is not helpful. I do not see into your filesystem. seems some dotfile is causing the issue.

**Comment by cpinfold** [comment]
Sorry not trying to make this difficult.

I have a leftover vim swapfile in my workflow directory: .class.WORKFLOW.php.swp

I changed: 
https://github.com/eventum/eventum/blob/40a6de2fc8c97ae58040434fa90237d5974b92d4/src/Extension/ExtensionLoader.php#L145
to
return '';

which matches the declaration return type:

https://github.com/eventum/eventum/blob/40a6de2fc8c97ae58040434fa90237d5974b92d4/src/Extension/ExtensionLoader.php#L142

and fixes the issue.

**Comment by glensc** [comment]
ok. reproduced by doing `touch config/workflow/.class.WORKFLOW.php.swp`

fixed with c4faabd.
